core/capabilities/ccip/oraclecreator/bootstrap.go

```
@@ -441,9 +453,15 @@ func (d *peerGroupDialer) closePeerGroup(endpointConfigDigest cciptypes.Bytes32)
 }
 
 func (d *peerGroupDialer) createPeerGroup(
+	ctx context.Context,
 	endpointConfigDigest cciptypes.Bytes32,
 	rmnHomeConfigDigest cciptypes.Bytes32,
 ) error {
+	if ctx.Err() != nil {
+		d.lggr.Debugw("peer group dialer closed, returning from createPeerGroup")
+		return nil
```

Should we return the actual err?

Also we could log the error

---

This isn't really how contexts should be used. You need to pass it into the code performing actual IO so that `ctx.Done()` can be checked within a select loop. So it also needs to be passed it into `GetRMNNodesInfo`.

With the pattern you have here, just check for a ctx Error before calling `createPeerGroup`.

---

GetRMNNodesInfo doesn't take a ctx because it actually doesn't do any I/O.

Unfortunately `Create` doesn't take in a ctx so we can't really plug the exact call that causes the race / panic.

The current `ctx.Err()` usage is somewhat uncommon but the alternative would be to do a `select` with a `case <-ctx.Done()` and `default:` which is effectively the same thing, no?

---

I'm also not sure about checking `Err()` directly instead of using `Done()`:
```
select {
case <-ctx.Done:
		d.lggr.Debugw("peer group dialer closed, returning from createPeerGroup")
		return nil
default:
        // the main logic here, or just fallthrough.
}
```